pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -363,13 +401,12 @@ void MultiTopicsConsumerImpl::closeAsync(ResultCallback callback) {
 
     auto self = shared_from_this();
     int numConsumers = 0;
-    consumers_.forEach(
```

Comment:
Because the consumers_ is removed in the handleSingleConsumerClose method, we can't directly use the consumers_ to traverse here.